src/LinkDotNet.NCronJob/Scheduler/CronScheduler.cs

```diff
@@ -78,6 +78,13 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
                         runningTasks.Add(task);
                         ScheduleJob(nextJob);
                     }
+                    else
```

Comment:
I will probably remove this with the approach from #51 with a semaphore.

Reply:
Ok, I'll also profile it to make sure I don't see those CPU spikes this intended on mitigating.

Reply:
@linkdotnet I've confirmed the CPU spike remains if I remove the Task.Delay(1). I tried a few other approaches but this seems to be the most effective. The CPU was spiking to 20% at one point while the scheduler was just sitting there waiting.

Reply:
I also confirmed this is an existing issue on the `main` branch, we can keep this fix on this PR or add separate PR just for this fix. 

To reproduce on main just add Job with a long async delay and a short interval like like this
```csharp
public class PrintHelloWorldJob : IJob
{
    public async Task RunAsync(JobExecutionContext context, CancellationToken token) => await Task.Delay(25000, token);
}

builder.Services.AddNCronJob(n => n
    .AddJob<PrintHelloWorldJob>(p =>
        p.WithCronExpression("*/5 * * * * *", timeZoneInfo: TimeZoneInfo.FindSystemTimeZoneById("Pacific Standard Time")))
);
```

Reply:
Hmm - Understandable. We might use this as our current debouncer. I would argue that even `Task.Delay(50)` should be good enough - anyway `Task.Delay(1)` is somewhat in the range of 1 to 16ms depending on the OS and arch (as they result in a different resolution for the timer).